OSDOCS-19990: Kueue Resource fair sharing (#113457)
openshift/openshift-docs: ai_workloads/kueue/admission-fair-sharing.adoc

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="admission-fair-sharing"]
= Admission fair sharing
:context: admission-fair-sharing

toc::[]

[role="_abstract"]
Use Admission fair sharing to fairly distribute workloads across LocalQueues that share a single ClusterQueue. 

> This feature balances workload admission by prioritizing workloads from tenants that have used fewer resources historically. It tracks usage over time with a configurable decay function and applies admission penalties when workloads are admitted.

Reviewer: ```suggestion
This feature balances workload admission by prioritizing workloads from local Queues that have used fewer resources historically. It tracks usage over time with a configurable decay function and applies admission penalties when workloads are admitted.
```

Author: changed.

> The shared ClusterQueue causes resource starvation between tenants, creating a high risk of resource starvation for the tenants. Admission fair sharing adresses this issue by meeting the following requirements:

Reviewer 2: "high risk of resource starvation" — redundant phrasing
Suggestion: "When multiple tenants share a single ClusterQueue, some tenants risk resource starvation. Admission fair sharing addresses this by..."

Author: changed.

Enforce multi-tenant fairness (business critical):: Ensure fair distribution of cluster resources across all tenants based on their usage history.  

Improve service predictability:: Guarantee each tenant gets a consistent share of resources, reducing latency spikes and preventing starvation.

> Enable scalable governance:: Use dynamic, usage-based allocation instead of complex static quotas.

Reviewer 2: I think this can be changed because Admission fair sharing does not replace quotas. It works alongside ClusterQueue quotas.. maybe something like:
"Complement static quotas with dynamic, usage-based admission ordering that adapts as tenant demand changes."

Author: changed.

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="admission-fair-sharing"]
= Admission fair sharing
:context: admission-fair-sharing

toc::[]

[role="_abstract"]
Use admission fair sharing to fairly distribute workloads across local Queues that share a single `ClusterQueue`. 

> This feature balances workload admission by prioritizing workloads from local Queues that have used fewer resources historically. It tracks usage over time with a configurable decay function and applies admission penalties when workloads are admitted.

Reviewer 3: As we're moving from being feature-focused to user-focused, consider revising line 12. Not essential for this PR, but a good thing to keep in mind, especially as we will start JTBD work soon. I don't know enough about this content to provide a good suggestion, but something like the following text might be a move in the right direction:

```suggestion
You can balance workload admission by prioritizing workloads from local Queues that have used fewer resources historically. With admission fair sharing, you can track usage over time with a configurable decay function and apply admission penalties when workloads are admitted.
```

Author: Changed.

> 

Reviewer 3: ```suggestion
```

Remove extra line, unless it's necessary here for some reason I can't decipher.

Author: removed line.
